### PR TITLE
Conversation node added to models

### PR DESCRIPTION
**Pull request by theTytoAlba**
This branch has implemented the node logic for the conversation tree.

* Every node has a title to describe their purpose.
* Every node has an intent. (Will be used to choose which child the parent will select to go).
* Every node has a parent. (Default parent is the root).
* Every node has a message that is going to be sent to the user when on that node.

Every conversation three has title and a root node.

(you can ignore the migration files, they are automatically created.)

### CODE REVIEW

File: bookbotproject/conversationtree/models.py [hunk 1]
```diff
@@ -3,9 +3,24 @@
 
 from django.db import models
 
+class ConversationNode(models.Model):
+    created = models.DateTimeField(auto_now_add=True)
+    #Every node has a title to describe their purpose.
+    title = models.CharField(max_length=100, blank=True)
+    # Every node has an intent. (Will be used to choose which child the parent will select to go).
+    intent = models.CharField(max_length=50, blank=True)
```
Comment (sevgicansalih) [inline bookbotproject/conversationtree/models.py]: intent field may not satisfy our needs. We can make it a longer CharField, like 100

File: bookbotproject/conversationtree/migrations/0004_auto_20171019_2008.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+    operations = [
+        migrations.RemoveField(
+            model_name='conversationnode',
+            name='name',
+        ),
+        migrations.AddField(
+            model_name='conversationnode',
+            name='intent',
+            field=models.CharField(blank=True, max_length=50),
```
Comment (sevgicansalih) [inline bookbotproject/conversationtree/migrations/0004_auto_20171019_2008.py:23]: Also this max_length field needs to be changed, I guess this is an autogenerated file, please dont forget to make necessary changes if you're going to change CharField length of "intent" field

**Review by sevgicansalih — COMMENTED** [review]
I think our node model deserves a few changes but well done and thanks Irmak !!

File: bookbotproject/conversationtree/models.py (hunk 1 above)
Comment (theTytoAlba) [reply bookbotproject/conversationtree/models.py]: I thought the intents to be shorter but you are right, lets make it 100 chars.

File: bookbotproject/conversationtree/migrations/0004_auto_20171019_2008.py (hunk 2 above)
Comment (theTytoAlba) [reply bookbotproject/conversationtree/migrations/0004_auto_20171019_2008.py:23]: Oh thank you, I was about to forget this but thanks to you I did that now. O:)

**Comment by theTytoAlba** [comment]
Please take another look @sevgicansalih

**Comment by sevgicansalih** [comment]
Thanks for your cooperation !! I'm merging and closing this request.

**Pull request merged**